packages/sx05re/emuelec/bin/setres.sh

```diff
@@ -4,77 +4,136 @@
 
 # Read the video output mode and set it for emuelec to avoid video flicking.
-MODE=`cat /sys/class/display/mode`;
+
+# This file sets the hdmi output and frame buffer to the argument in pixel width.
+# Allowed argument example ./setres.sh 1080p60hz <-- For height 1080 pixels.
+
+# set -x #echo on
+
+# 1080p60hz
+# 1080i60hz
+# 720p60hz
+# 720p50hz
+# 480p60hz
+# 480cvbs
+# 576p50hz
+# 1080p50hz
+# 1080i50hz
+# 576cvbs
+
 BPP=32
+HZ=60
 
-if [ -e /ee_s905 ]; then
-case "$MODE" in
-"480p60hz")
-	fbset -fb /dev/fb0 -g 720 480 720 960 $BPP
-	;;
-"576p50hz")
-	fbset -fb /dev/fb0 -g 720 576 720 1152 $BPP
-	;;
-"720p60hz")
-	fbset -fb /dev/fb0 -g 1280 720 1280 1440 $BPP
-	;;
-"720p50hz")
-	echo 720p60hz > /sys/class/display/mode
-	fbset -fb /dev/fb0 -g 1280 720 1280 1440 $BPP
-	;;
-"1080p60hz")
-	fbset -fb /dev/fb0 -g 1920 1080 1920 2160 $BPP
-	;;
-"1080i60hz")
-	fbset -fb /dev/fb0 -g 1920 1080 1920 2160 $BPP
-	;;
-"1080i50hz")
-	echo 1080i60hz > /sys/class/display/mode
-	fbset -fb /dev/fb0 -g 1920 1080 1920 2160 $BPP
-	;;
-"1080p50hz")
-	echo 1080p60hz > /sys/class/display/mode
-	fbset -fb /dev/fb0 -g 1920 1080 1920 2160 $BPP
-	;;
-"1280x1024p60hz")
-	fbset -fb /dev/fb0 -g 1280 1024 1280 2048 $BPP
-	echo 0 0 1279 1023 > /sys/class/graphics/fb0/free_scale_axis
-	echo 0 0 1279 1023 > /sys/class/graphics/fb0/window_axis
-	echo 0x10001 > /sys/class/graphics/fb0/free_scale
-	;;
-"1024x768p60hz")
-	fbset -fb /dev/fb0 -g 1024 768 1024 1536 $BPP
-	echo 0 0 1023 767 > /sys/class/graphics/fb0/free_scale_axis
-	echo 0 0 1023 767 > /sys/class/graphics/fb0/window_axis
-	echo 0x10001 > /sys/class/graphics/fb0/free_scale
-	;;
-"640x480p60hz")
-	fbset -fb /dev/fb0 -g 640 480 640 960 $BPP
-	echo 0 0 639 479 > /sys/class/graphics/fb0/free_scale_axis
-	echo 0 0 639 479 > /sys/class/graphics/fb0/window_axis
-	echo 0x10001 > /sys/class/graphics/fb0/free_scale
-	;;
-"480cvbs")
-	fbset -fb /dev/fb0 -g 1280 960 1280 1920 $BPP
-	fbset -fb /dev/fb1 -g $BPP $BPP $BPP $BPP $BPP
-	echo 0 0 1279 959 > /sys/class/graphics/fb0/free_scale_axis
-	echo 30 10 669 469 > /sys/class/graphics/fb0/window_axis
-	echo 640 > /sys/class/graphics/fb0/scale_width
-	echo 480 > /sys/class/graphics/fb0/scale_height
-	echo 0x10001 > /sys/class/graphics/fb0/free_scale
-	;;
-"576cvbs")
-	fbset -fb /dev/fb0 -g 1280 960 1280 1920 $BPP
-	fbset -fb /dev/fb1 -g $BPP $BPP $BPP $BPP $BPP
-	echo 0 0 1279 959 > /sys/class/graphics/fb0/free_scale_axis
-	echo 35 20 680 565 > /sys/class/graphics/fb0/window_axis
-	echo 720 > /sys/class/graphics/fb0/scale_width
-	echo 576 > /sys/class/graphics/fb0/scale_height
-	echo 0x10001 > /sys/class/graphics/fb0/free_scale
-	;;
+MODE=$1
+
+[ -z "$MODE" ] && MODE=`cat /sys/class/display/mode`;
+
+case $MODE in
+	*p*) H=$(echo $MODE | cut -d'p' -f 1) ;;
+	*i*) H=$(echo $MODE | cut -d'i' -f 1) ;;
+	*cvbs*) H=$(echo $MODE | cut -d'c' -f 1) ;;
+	*hz*) HZ=${i:(-4):2} ;;
 esac
 
-fi 
+if [ ! -n "$H" ]; then
+	H=$(echo $H | cut -d'x' -f 2)
+fi
```

Comment:
Fix for issue getting height from 1280x1024, 1024x768:
https://github.com/EmuELEC/EmuELEC/pull/694#issuecomment-903704422

I haven't tested it properly, but I dont see why it shouldn't work!

Reply:
Hi, i'm not sure because i can't test it now. But i have a doubt on this formula W=$(($W*5/4)). It's just an advice i dont know  how to do that

Reply:
> Hi, i'm not sure because i can't test it now. But i have a doubt on this formula W=$(($W*5/4)). It's just an advice i dont know how to do that

Was a typo, should be fixed now.